src/test/java/ysoserial/test/payloads/Struts2JasperReportsTest.java

```diff
@@ -0,0 +1,47 @@
+package ysoserial.test.payloads;
+
+import java.util.HashMap;
+
+import com.opensymphony.xwork2.ActionContext;
+import com.opensymphony.xwork2.config.Configuration;
+import com.opensymphony.xwork2.config.ConfigurationManager;
+import com.opensymphony.xwork2.config.providers.XWorkConfigurationProvider;
+import com.opensymphony.xwork2.inject.Container;
+
+import ysoserial.Deserializer;
+import ysoserial.test.CustomDeserializer;
+
+public class Struts2JasperReportsTest extends CommandExecTest implements CustomDeserializer {
+
+	@Override
+	public Class<?> getCustomDeserializer() {
+		return StrutsJasperReportsDeserializer.class;
+	}
+	
+    /**
+     * need to use a custom deserializer so that the action context gets set in the isolated class
+     *
+     * @author sciccone
+     *
+     */
+    public static final class StrutsJasperReportsDeserializer extends Deserializer {
+        
+		public StrutsJasperReportsDeserializer(byte[] bytes) {
+			super(bytes);
+		}
+    	
+        @Override
+        public Object call () throws Exception {
```

Comment:
Is there a reason this context setup needs to be in a custom `Deserializer` class as opposed to just being in beginning of the payload `getObject()` implementation?

Reply:
I have tried to put it within the `getObject` implementation, and if I run the `main` in the payload class works. But, when I run the complete tests, it seems the deserialisation appears in a different thread where the `ActionContext` has not been initialised. So, I created a custom `Deserializer` for doing that.

Reply:
Hi @frohoff, do you have updates on merging this branch? Are there any problems?

Reply:
No problems, just been super busy. Still hope to review more and merge. Thanks.